server/chat-plugins/datasearch.ts

```diff
@@ -1725,20 +1727,22 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
 				if (alts.gens[String(move.gen)]) continue;
 				if (Object.values(alts.gens).includes(false) && alts.gens[String(move.gen)] !== false) continue;
 			}
-			for (const recoveryType in alts.other) {
-				let hasRecovery = false;
-				if (recoveryType === "recovery") {
-					hasRecovery = !!move.drain || !!move.flags.heal;
-				} else if (recoveryType === "zrecovery") {
-					hasRecovery = (move.zMove?.effect === 'heal');
-				}
-				if (hasRecovery === alts.other[recoveryType]) {
-					matched = true;
-					break;
-				}
-			}
+			if (!zrecoveryUndefined || !recoveryUndefined) {
+			  for (const recoveryType in alts.other) {
+				 let hasRecovery = false;
+				 if (recoveryType === "recovery") {
+					 hasRecovery = !!move.drain || !!move.flags.heal;
+				 } else if (recoveryType === "zrecovery") {
+					 hasRecovery = (move.zMove?.effect === 'heal');
+				 }
+				 if (hasRecovery === alts.other[recoveryType]) {
+					 matched = true;
+					 break;
+				 }
+			    }
+		    }
```

Comment:
```suggestion
			if (!zrecoveryUndefined || !recoveryUndefined) {
				for (const recoveryType in alts.other) {
					let hasRecovery = false;
					if (recoveryType === "recovery") {
						hasRecovery = !!move.drain || !!move.flags.heal;
					} else if (recoveryType === "zrecovery") {
						hasRecovery = (move.zMove?.effect === 'heal');
					}
					if (hasRecovery === alts.other[recoveryType]) {
						matched = true;
						break;
					}
				}
			}
```
This isn't indented properly.